### ISSUE TITLE
Running on macOS Big Sur 11.5 causes abort for freeing unallocated pointer when any key on controller is played.

### ISSUE THREAD

**Issue by mrburrito** [xobs/midi-to-keypress#9]
I found this project while I was looking for a way to use my MIDI keyboard (Amazon DeepComposer - `AKM322`) with FFXIV's performer mode on my mac. Super excited to get it working!

It seems to start up, recognize, and connect to my keyboard but as soon as I play a note, it bombs. Tried several different notes on the keyboard and they all end up with the same result.

### Output

```
$ ./miditran
Connection established to AKM322
miditran(18726,0x70000dbbe000) malloc: *** error for object 0x7feacfd13100: pointer being freed was not allocated
miditran(18726,0x70000dbbe000) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6

$ ./miditran --version
Midi Perform 0.2.9
```

### Hardware
- 2018 MacBook Pro, 15"
- 6-Core Core i9
- 32GB DDR4 2400
- Radeon Pro 560X 4GB
- Intel UHD Graphics 630 1536MB
- CalDigit TS3+ Dock
- Amazon DeepComposer USB Keyboard

DeepComposer is connected via USB to the CalDigit dock, which is connected via Thunderbolt 3 to the mac

### Software

- macOS Big Sur 11.5

**Comment by mrburrito** [comment]
I definitely downloaded the 0.3.7 bundle, but the version is showing 0.2.9. Not sure if that's intentional for the mac version or if it might be a compile issue?

**Comment by mrburrito** [comment]
Interesting. The compiled binary generates the error above, but if I download the source and run it with `cargo`, it works as intended.

Even with that, I uncommented your `println` statement for key event output and saw that the app is doing what it's supposed to do -- generating the correct KeyDown/KeyUp when I hit keys on the keyboard -- but they aren't recognized in game. Is there anything I need to do in System Prefs/Security & Privacy to grant access?

**Comment by xobs** [comment]
The `0.2.9` is probably due to me hardcoding the version number in https://github.com/xobs/midi-to-keypress/blob/master/src/main.rs#L38 -- there wasn't a method to derive the version number before, but now there is. I will update it.

Are keys recognized in any other program? For example, in a text editor? What about in Terminal? According to https://github.com/asweigart/pyautogui/issues/247#issuecomment-437668855 you may need to add Terminal to the list. Please try it and see if that fixes it.

**Comment by mrburrito** [comment]
Yeah, I saw the hardcoded version number in there when I was looking at stuff. Trying to add it to Accessibilty now -- no, not recognized in any programs.

**Comment by mrburrito** [comment]
That was it! Just need to figure out why it only works via cargo.

**Comment by mrburrito** [comment]
I did a `cargo build` and running from that binary, everything works. Might just be that it needs to be recompiled under Big Sur?

**Comment by mrburrito** [comment]
Thanks for this by the way! It's fantastic! Now I just need to figure out the right keymapping. :)

**Comment by xobs** [comment]
I fixed the version number in 605af14ba9215cf802f869963ecaf2b920c5ce91 so that will be correct in future versions. Thanks for pointing that out.

What if you do `cargo run`? Does it work if you do `cargo run --release`? I wonder if there's something that's getting optimized away. Right now the Github Action for building on Mac is:

```
$ cargo build --all --release && strip target/release/miditran && (cd target/release && zip ../../miditran-macos.zip miditran)
```

I set the macOS version to 10.10, which should provide better compatibility. Can you pelase see if the v0.3.8 release solves any of the issues you've described?

**Comment by mrburrito** [comment]
If I run with `cargo run --release` or with the binary after running the build command, it registers a single KeyDown event and then seems to hang. (I uncommented the println for the keypress in main.rs)

```
./miditran
Connection established to AKM322
Found note mapping: NoteMapping { note: D4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('w'))], off: [KeyUp(Layout('w'))] } for event NoteOn, running sequence [NoteMod(None), KeyDown(Layout('w'))]
```

If I just do `cargo run`, it seems to work properly.

```
cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/miditran`
Connection established to AKM322
Found note mapping: NoteMapping { note: C4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('q'))], off: [KeyUp(Layout('q'))] } for event NoteOn, running sequence [NoteMod(None), KeyDown(Layout('q'))]
Found note mapping: NoteMapping { note: C4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('q'))], off: [KeyUp(Layout('q'))] } for event NoteOff, running sequence [KeyUp(Layout('q'))]
qFound note mapping: NoteMapping { note: D4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('w'))], off: [KeyUp(Layout('w'))] } for event NoteOn, running sequence [NoteMod(None), KeyDown(Layout('w'))]
wFound note mapping: NoteMapping { note: D4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('w'))], off: [KeyUp(Layout('w'))] } for event NoteOff, running sequence [KeyUp(Layout('w'))]
Found note mapping: NoteMapping { note: E4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('e'))], off: [KeyUp(Layout('e'))] } for event NoteOn, running sequence [NoteMod(None), KeyDown(Layout('e'))]
eFound note mapping: NoteMapping { note: E4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('e'))], off: [KeyUp(Layout('e'))] } for event NoteOff, running sequence [KeyUp(Layout('e'))]
Found note mapping: NoteMapping { note: F4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('r'))], off: [KeyUp(Layout('r'))] } for event NoteOn, running sequence [NoteMod(None), KeyDown(Layout('r'))]
rFound note mapping: NoteMapping { note: F4, channel: 0, instrument_name: None, on: [NoteMod(None), KeyDown(Layout('r'))], off: [KeyUp(Layout('r'))] } for event NoteOff, running sequence [KeyUp(Layout('r'))]
```

**Comment by mrburrito** [comment]
Tried running a `cargo clean` first and exporting `MACOSX_DEPLOYMENT_TARGET` as both `10.10` and `11.0` and got the same results (single keypress then hanging). Both compiling and with `run --release`.
